### PR TITLE
Fix optional typehint for array and container functions

### PR DESCRIPTION
**Pull request by xoiga123**
[Trello task](https://trello.com/c/coS9qGNO/1392-correct-use-of-optional-type-hint)

This PR also fixes some minor typos and obvious errors

### CODE REVIEW

File: ivy/array/sorting.py [hunk 1]
```diff
@@ -102,7 +105,7 @@ def searchsorted(
         /,
         *,
         side="left",
-        sorter=None,
+        sorter: Optional = None,
```
Comment (AnnaTz) [inline ivy/array/sorting.py:108]: Can this be just Optional? What else could it be except for None?

File: ivy/container/linear_algebra.py [hunk 2]
```diff
@@ -2379,13 +2379,10 @@ def tensorsolve(
...
-        axes: Union[int, Tuple[List[int], List[int]]] = None,
+        axes: Optional[Union[int, Tuple[List[int], List[int]]]] = None,
     ) -> ivy.Container:
-        return self.tensorsolve(
-            self,
-            x2,
-            axes=axes,
-        )
+        # TODO: implement this function
+        pass
```
Comment (AnnaTz) [inline ivy/container/linear_algebra.py:2385]: Is there no `tensorsolve`?

File: ivy/container/sorting.py [hunk 3]
```diff
@@ -340,7 +340,7 @@ def static_searchsorted(
         /,
         *,
         side="left",
-        sorter=None,
+        sorter: Optional = None,
```
Comment (AnnaTz) [inline ivy/container/sorting.py:343]: Same as for Array.

File: ivy/container/sorting.py [hunk 4]
```diff
@@ -374,7 +374,7 @@ def searchsorted(
...
```
Comment (AnnaTz) [inline ivy/container/sorting.py:377]: Same.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy/container/linear_algebra.py (hunk 2 above)
Comment (AnnaTz) [reply ivy/container/linear_algebra.py:2385]: https://github.com/unifyai/ivy/blob/b98068a7e60162f83e1003b18c52314494addb91/ivy/functional/ivy/linear_algebra.py#L2141

Comment (xoiga123) [reply ivy/container/linear_algebra.py]: The old implementation was wrong, and I was scared this PR may go out of scope so I left it as a TODO :sweat_smile:

Comment (AnnaTz) [reply ivy/container/linear_algebra.py]: I didn't think it through when I saw you deleted the function call tbh 😅 You can leave it as it is now, no need to add docstrings.

**Pull request merged**